app/Models/Characters/Monster.php

```diff
@@ -34,4 +40,142 @@ public function getProficiencyBonusAttribute()
+    }
+
+    function getAcrobaticsAttribute()
+    {
+        $score = $this->dexterity_modifier;
+
+        return $score + $this->getSkillBonus('acrobatics');
+    }
+
+    function getSleightOfHandAttribute()
+    {
+        $score = $this->dexterity_modifier;
+
+        return $score + $this->getSkillBonus('sleight_of_hand');
+    }
+
+    function getStealthAttribute()
+    {
+        $score = $this->dexterity_modifier;
+
+        return $score + $this->getSkillBonus('stealth');
+    }
+
+    function getArcanaAttribute()
+    {
+        $score = $this->intelligence_modifier;
+
+        return $score + $this->getSkillBonus('arcana');
+    }
+
+    function getHistoryAttribute()
+    {
+        $score = $this->intelligence_modifier;
+
+        return $score + $this->getSkillBonus('history');
+    }
+
+    function getInvestigationAttribute()
+    {
+        $score = $this->intelligence_modifier;
+
+        return $score + $this->getSkillBonus('investigation');
+    }
+
+    function getNatureAttribute()
+    {
+        $score = $this->intelligence_modifier;
+
+        return $score + $this->getSkillBonus('nature');
+    }
+
+    function getReligionAttribute()
+    {
+        $score = $this->intelligence_modifier;
+
+        return $score + $this->getSkillBonus('religion');
+    }
+
+    function getAnimalHandlingAttribute()
+    {
+        $score = $this->wisdom_modifier;
+
+        return $score + $this->getSkillBonus('animal_handling');
+    }
+
+    function getInsightAttribute()
+    {
+        $score = $this->wisdom_modifier;
+
+        return $score + $this->getSkillBonus('insight');
+    }
+
+    function getMedicineAttribute()
+    {
+        $score = $this->wisdom_modifier;
+
+        return $score + $this->getSkillBonus('medicine');
+    }
+
+    function getPerceptionAttribute()
+    {
+        $score = $this->wisdom_modifier;
+
+        return $score + $this->getSkillBonus('perception');
+    }
+
+    function getSurvivalAttribute()
+    {
+        $score = $this->wisdom_modifier;
+
+        return $score + $this->getSkillBonus('survival');
+    }
+
+    function getDeceptionAttribute()
+    {
+        $score = $this->charisma_modifier;
+
+        return $score + $this->getSkillBonus('deception');
+    }
+
+    function getIntimidationAttribute()
+    {
+        $score = $this->charisma_modifier;
+
+        return $score + $this->getSkillBonus('intimidation');
+    }
+
+    function getPerformanceAttribute()
+    {
+        $score = $this->charisma_modifier;
+
+        return $score + $this->getSkillBonus('performance');
+    }
+
+    function getPersuasionAttribute()
+    {
+        $score = $this->charisma_modifier;
+
+        return $score + $this->getSkillBonus('persuasion');
+    }
+
+    function getSkillBonus(string $skill_name)
+    {
+        $skill = $this->skills()->where('name', '=', $skill_name)->first();
+
+        return $skill ? $skill->pivot->bonus : 0;
```

Comment:
Is this specific to Monsters?

If is applicable to Heros/Characters or any other model, you could have it as a trait, would also keep the model cleaner!

Reply:
Scores are calculated quite differently for monsters Vs characters, but thinking about it this could all be a trait and monsters will just need to override the getSkillBonus method